pkg/scalers/predictkube_scaler.go

```diff
@@ -0,0 +1,463 @@
+	validate := validator.New()
+	meta := predictKubeMetadata{}
+
+	if val, ok := config.TriggerMetadata["query"]; ok {
+		if len(val) == 0 {
+			return nil, fmt.Errorf("no query given")
+		}
+
+		meta.query = val
+	} else {
+		return nil, fmt.Errorf("no query given")
+	}
+
+	if val, ok := config.TriggerMetadata["prometheusAddress"]; ok {
+		err = validate.Var(val, "url")
+		if err != nil {
+			return nil, fmt.Errorf("invalid prometheusAddress")
+		}
+
+		meta.prometheusAddress = val
+	} else {
+		return nil, fmt.Errorf("no prometheusAddress given")
+	}
+
+	if val, ok := config.TriggerMetadata["predictHorizon"]; ok {
+		meta.predictHorizon, err = str2duration.ParseDuration(val)
+		if err != nil {
+			return nil, fmt.Errorf("predictHorizon parsing error %s", err.Error())
+		}
+	} else {
+		return nil, fmt.Errorf("no predictHorizon given")
+	}
+
+	if val, ok := config.TriggerMetadata["queryStep"]; ok {
+		meta.stepDuration, err = str2duration.ParseDuration(val)
+		if err != nil {
+			return nil, fmt.Errorf("queryStep parsing error %s", err.Error())
+		}
+	} else {
+		return nil, fmt.Errorf("no queryStep given")
+	}
+
+	if val, ok := config.TriggerMetadata["historyTimeWindow"]; ok {
+		meta.historyTimeWindow, err = str2duration.ParseDuration(val)
+		if err != nil {
+			return nil, fmt.Errorf("historyTimeWindow parsing error %s", err.Error())
+		}
+	} else {
+		return nil, fmt.Errorf("no historyTimeWindow given")
+	}
+
+	if val, ok := config.TriggerMetadata["threshold"]; ok {
+		meta.threshold, err = strconv.ParseInt(val, 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("threshold parsing error %s", err.Error())
+		}
+	} else {
+		return nil, fmt.Errorf("no threshold given")
+	}
+
+	if val, ok := config.TriggerMetadata["metricName"]; ok {
+		if len(val) == 0 {
+			return nil, fmt.Errorf("no metric name given")
+		}
+
+		err = validate.Var(val, "ascii")
+		if err != nil {
+			return nil, fmt.Errorf("invalid apiKey")
+		}
+
+		meta.metricName = val
+	} else {
+		return nil, fmt.Errorf("no metric name given")
+	}
+
+	meta.scalerIndex = config.ScalerIndex
+
+	if val, ok := config.AuthParams["apiKey"]; ok {
+		err = validate.Var(val, "jwt")
+		if err != nil {
+			return nil, fmt.Errorf("invalid apiKey")
+		}
+
+		meta.apiKey = val
+	} else {
+		return nil, fmt.Errorf("no api key given")
+	}
+
+	return &meta, nil
+}
+
+func (pks *PredictKubeScaler) ping(ctx context.Context) (err error) {
+	_, err = pks.api.Runtimeinfo(ctx)
+	return err
+}
+
+// initPredictKubePrometheusConn init prometheus client and setup connection to API
+func (pks *PredictKubeScaler) initPredictKubePrometheusConn() (err error) {
+	pks.transport = http_transport.NewTransport(&libs.HTTPTransport{
+		MaxIdleConnDuration: 10,
+		ReadTimeout:         time.Second * 15,
+		WriteTimeout:        time.Second * 15,
+	})
+
+	pks.prometheusClient, err = api.NewClient(api.Config{
+		Address:      pks.metadata.prometheusAddress,
+		RoundTripper: pks.transport,
+	})
```

Comment:
Should we support at least the same authentication methods that we already support for prometheus scaler?

Reply:
Yeah, it should support all the auth that does the current prometheus scaler. I think that it might be worth to use some generic way how to parse prometheus related metadata in both scalers (prometheus and this one). So we don't have too much code duplication.

Reply:
Currently your request is in progress.

Reply:
fixed